plugins/modules/keycloak_realm_key.py

```diff
@@ -0,0 +1,457 @@
+                description:
+                    - The private key as an ASCII string. Contents of the key must match O(config.algorithm)
+                      and O(provider_id).
+                    - Please note that the module cannot detect whether the private key specified differs from the
+                      current state's private key. Use O(force=true) to force the module to update the private key
+                      if you expect it to be updated.
+                required: true
+                type: str
+            certificate:
+                description:
+                    - A certificate signed with the private key as an ASCII string. Contents of the
+                      key must match O(config.algorithm) and O(provider_id).
+                    - If you want Keycloak to automatically generate a certificate using your private key
+                      then set this to an empty string.
+                required: true
+                type: str
+notes:
+    - Current value of the private key cannot be fetched from Keycloak.
+      Therefore comparing its desired state to the current state is not
+      possible.
+    - If certificate is not explicitly provided it will be dynamically created
+      by Keycloak. Therefore comparing the current state of the certificate to
+      the desired state (which may be empty) is not possible.
+    - Due to the private key and certificate options the module is
+      B(not fully idempotent). You can use O(force=true) to force the module
+      to always update if you know that the private key might have changed.
+
+extends_documentation_fragment:
+    - community.general.keycloak
+    - community.general.attributes
+
+author:
+    - Samuli Seppänen (@mattock)
+'''
+
+EXAMPLES = '''
+- name: Manage Keycloak realm key (certificate autogenerated by Keycloak)
+  community.general.keycloak_realm_key:
+    name: custom
+    state: present
+    parent_id: master
+    provider_id: rsa
+    auth_keycloak_url: http://localhost:8080/auth
+    auth_username: keycloak
+    auth_password: keycloak
+    auth_realm: master
+    config:
+      private_key: "{{ private_key }}"
+      enabled: true
+      active: true
+      priority: 120
+      algorithm: RS256
+- name: Manage Keycloak realm key and certificate
+  community.general.keycloak_realm_key:
+    name: custom
+    state: present
+    parent_id: master
+    provider_id: rsa
+    auth_keycloak_url: http://localhost:8080/auth
+    auth_username: keycloak
+    auth_password: keycloak
+    auth_realm: master
+    config:
+      private_key: "{{ private_key }}"
+      certificate: "{{ certificate }}"
+      enabled: true
+      active: true
+      priority: 120
+      algorithm: RS256
+'''
+
+RETURN = '''
+msg:
+    description: Message as to what action was taken.
+    returned: always
+    type: str
+
+end_state:
+    description: Representation of the keycloak_realm_key after module execution.
+    returned: on success
+    type: complex
```

Comment:
This changed back to `complex`.
```suggestion
    type: dict
```